### PR TITLE
Task1.Lesson7

### PR DESCRIPTION
**Pull request by NeverovDS**
Implementation CRUD

### CODE REVIEW

File: web_serverT1V2.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+)
+
+
+type  RequestInfo struct {
+	Host string `json:"host"`
+	User_agent string `json:"user_agent"`
```
Comment (imatsiash-ib) [inline web_serverT1V2.go]: Please use CamelCase.

Comment (imatsiash-ib) [reply web_serverT1V2.go]: Did you use `go fmt` ?

File: web_serverT1V2.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+func RequestHandler(w http.ResponseWriter, r *http.Request)  {
+	w.Header().Set("Content-Type", "application/json")
+	request := RequestInfo{
+		Host:r.Host,
+		User_agent:r.UserAgent(),
+		Request_uri:r.RequestURI,
+		Headers:r.Header,
+		}
+
+	requestJson, _ := json.Marshal(request)
```
Comment (imatsiash-ib) [inline web_serverT1V2.go]: Please do not ignore error.
In case of error you should return it back to client.

Comment (NeverovDS) [reply web_serverT1V2.go]: +

**Comment by burov** [comment]
Please solve task in a right way

File: web_serverT1V2.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	request := RequestInfo{
+		Host:       r.Host,
+		UserAgent:  r.UserAgent(),
+		RequestUri: r.RequestURI,
+		Headers:    r.Header,
+	}
+
+	requestJson, err := json.Marshal(request)
+	if err != nil {
+		errors.New("Shit happens.You are fucked up")
```
Comment (imatsiash-ib) [inline web_serverT1V2.go]: It’s a bad idea to return such a message to the user.

Comment (imatsiash-ib) [reply web_serverT1V2.go]: In case of error you should return error back to the client and set status code to error code.

Comment (NeverovDS) [reply web_serverT1V2.go]: +

File: web_serverT1V2.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	request := RequestInfo{
+		Host:       r.Host,
+		UserAgent:  r.UserAgent(),
+		RequestUri: r.RequestURI,
+		Headers:    r.Header,
+	}
+
+	requestJson, err := json.Marshal(request)
+	if err != nil {
+		http.Error(w,"something went wrong with parsing",http.StatusBadRequest)
```
Comment (imatsiash-ib) [inline web_serverT1V2.go]: Do not forget about `return` here.

**Pull request closed without merging**